app/dlc-suredbits-client/src/main/scala/org/bitcoins/sbclient/SbClient.scala

```diff
@@ -0,0 +1,137 @@
+    implicit val ec: ExecutionContextExecutor = m.executionContext
+
+    Http()
+      .singleRequest(HttpRequest(uri = uri))
+      .flatMap(
+        response =>
+          response.entity.dataBytes
+            .runFold(ByteString.empty)(_ ++ _)
+            .map(payload => payload.decodeString(ByteString.UTF_8))
+      )
+  }
+
+  def restCall(uri: Uri)(implicit system: ActorSystem): Future[JsValue] = {
+    rawRestCall(uri).map(Json.parse)(system.dispatcher)
+  }
+
+  import org.bitcoins.commons.serializers.JsonReaders.lnInvoiceReads
+  implicit val invoiceAndDataResponseReads: Reads[InvoiceAndDataResponse] =
+    Json.reads[InvoiceAndDataResponse]
+
+  def request(
+      exchange: Exchange,
+      tradingPair: TradingPair,
+      requestType: RequestType,
+      endpoint: String = "https://test.api.suredbits.com/dlc/v0")(
+      implicit system: ActorSystem): Future[InvoiceAndDataResponse] = {
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+    val uri =
+      s"$endpoint/${exchange.toLongString}/${tradingPair.toLowerString}/${requestType.requestString}"
+    restCall(uri)
+      .map(_.validate[InvoiceAndDataResponse])
+      .flatMap {
+        case JsSuccess(response, _) => Future.successful(response)
+        case JsError(error) =>
+          Future.failed(
+            new RuntimeException(
+              s"Unexpected error when parsing response: $error"))
+      }
+  }
+
+  def makePayment(
+      eclairApi: EclairApi,
+      invoice: LnInvoice,
+      timeout: FiniteDuration = 5.seconds)(
+      implicit ec: ExecutionContext): Future[PaymentPreimage] = {
+    for {
+      payment <- eclairApi.payAndMonitorInvoice(invoice,
+                                                externalId = None,
+                                                interval = timeout / 10,
+                                                maxAttempts = 10)
+    } yield payment.status match {
+      case OutgoingPaymentStatus.Succeeded(preImage, _, _, _) => preImage
+      case OutgoingPaymentStatus.Failed(errs) =>
+        val errMsgs = errs.map(_.failureMessage).mkString(",\n")
+        throw new RuntimeException(s"Payment failed: $errMsgs")
+      case OutgoingPaymentStatus.Pending =>
+        throw new IllegalStateException("This should not be possible.")
+    }
+  }
+
+  def decryptData(data: String, preImage: PaymentPreimage): String = {
+    AesCrypt.decrypt(AesEncryptedData.fromValidBase64(data),
+                     AesKey.fromValidBytes(preImage.bytes)) match {
+      case Left(err) => throw err
+      case Right(decrypted) =>
+        decrypted.decodeUtf8 match {
+          case Right(decodedStr) => decodedStr
+          case Left(err)         => throw err
+        }
+    }
+  }
+
+  def requestAndPay(
+      exchange: Exchange,
+      tradingPair: TradingPair,
+      requestType: RequestType,
+      eclairApi: EclairApi,
+      endpoint: String = "https://test.api.suredbits.com/dlc/v0")(
```

Comment:
maybe make this configurable in the `reference.conf` file?

Reply:
I think at this layer of abstraction, we should take in a `String` (likely without a default value in the future) but at some point, I think it would make sense to have the layer above this read an endpoint from a conf